src/azure-cli/azure/cli/command_modules/iot/custom.py

```diff
@@ -425,13 +426,13 @@ def iot_hub_create(cmd, client, hub_name, resource_group_name, location=None,
     if fileupload_storage_container_name and not fileupload_storage_connectionstring:
         raise RequiredArgumentMissingError('Please mention storage connection string.')
     identity_based_file_upload = fileupload_storage_authentication_type and fileupload_storage_authentication_type.lower() == AuthenticationType.IdentityBased.value
-    if not identity_based_file_upload and not fileupload_storage_connectionstring and fileupload_storage_container_name:
-        raise RequiredArgumentMissingError('Key-based authentication requires a connection string.')
-    if identity_based_file_upload and not fileupload_storage_container_uri:
-        raise RequiredArgumentMissingError('Identity-based authentication requires a storage container uri (--fileupload-storage-container-uri, --fcu).')
     if not identity_based_file_upload and fileupload_storage_identity:
         raise RequiredArgumentMissingError('In order to set a fileupload storage identity, please set file upload storage authentication (--fsa) to IdentityBased')
-
+    if identity_based_file_upload or fileupload_storage_identity:
+        if fileupload_storage_identity == SYSTEM_ASSIGNED_IDENTITY and not system_identity:
```

Comment:
nit, the next `if` statement first checks if `fileupload_storage_identity` is a truthy value and then does the value comparison....but this `if` statement skips the truthy check.

Reply:
Ok - thanks for pointing this out. Just realized this is because a user can set `--fsa` to enable identityBased file upload - but realized there's a gap in the check here because technically `--fsi` is *not* required (left empty it assumes system assigned identity). I'll update the logic for this.